### DISCUSSION TITLE
Python: use a var in a query

### DISCUSSION THREAD

**Discussion by wdennis** [duckdb/duckdb#12219] [Q&A] [answered]
Hi all, DuckDB (and Python) newb here. I'm trying to use a variable containing a table name (str) in a query, as so:

`con.execute("SELECT * FROM ?", [table_to_analyze]).fetchone()`

but I'm getting an error as so: `duckdb.duckdb.ParserException: Parser Error: syntax error at or near "?"`

How may I do this seemingly simple thing?

**Comment by Tishj** [comment, marked as answer]
Use an f string in Python
Prepared parameters (`?`) can be used for expressions, not for table names

**Reply by wdennis** [reply]
Thanks, `con.execute(f"SELECT * FROM {table_to_analyze}").fetchone()` does the trick :)

**Reply by Tishj** [reply]
If you're using the python API, you could also use `con.table(table_to_analyze)`, followed with a `.select("a, b, c")` if necessary

**Comment by wdennis** [comment]
Ah, good to know - thanks!
